science/gnuradio/Portfile

```diff
@@ -1,458 +1,317 @@
-
-    patchfiles-append \
-        patch-C_CXX_STANDARD.devel.diff
-
-    # patch to fix use of GNUCXX to instead be for both GNU
-    # and Clang compilers
-
-    patchfiles-append \
-        patch-fix-GNUCXX.devel.diff
+    set GR_VERSION_INFO "Release3.7"
 
-    # patch to fix Boost #include
-
-    patchfiles-append \
-        patch-gnuradio-runtime_lib_buffer.cc.boost_include_fix.devel.diff
-
-    # patch to fix type usage of MP package
-
-    patchfiles-append \
-        patch-fix-mp.devel.diff
-
-    # patch to fix installing CTRLPORT Python apps location
-
-    patchfiles-append \
-        patch-fix-ctrlport-python-app-install-location.devel.diff
-
-    # patch to fix FindLog2cpp.cmake script to correct cache LOG4CPP_FOUND
-
-    patchfiles-append \
-        patch-fix-FindLog4cpp.devel.diff
+    # still require cppunit for testing; NOTE: cppunit is checked for
+    # during configure, so we need it to be in depends_lib or
+    # depends_build to be used correctly. Choose the latter since it's
+    # not required for runtime; just for build/test.
+    depends_build-append \
+        port:cppunit
 
-    # patch to fix unused RPC variables
+    github.livecheck.regex  {(3.7.[0-9.]+)}
 
-    patchfiles-append \
-        patch-fix-rpc-unused-vars.devel.diff
+}
 
-    # temporary patch to fix using Thrift 0.9.2 through 0.12.0
+subport gnuradio-next {
 
-    patchfiles-append \
-        patch-fixes_for-thrift.devel.diff
+    name gnuradio-next
 
-} else {
+    github.setup gnuradio gnuradio 5099fd9a492face0facfc4a8a6923d5ce76b7b31
+    version   20191003-[string range ${github.version} 0 7]
+    checksums rmd160 3f27981c0fbc2abf197fb59f7054b774676d47f5 \
+              sha256 101cd40bd9765ba2a2264f44600b8a7088740a136e7c5457b783c8e87626da3b \
+              size   3369676
+    revision  3
 
-    # patches for next
+    long_description    ${description}: \
+        This port is kept up with the GNU Radio GIT 'master' \
+        branch (as of 2018-09-20), which is typically updated \
+        daily to weekly.  This version of GNU Radio represents \
+        the next major release of GNU Radio, and hence its API \
+        is likely to be different than that provided by gnuradio. \
+        This port may or not compile \
+        or function correctly, as it represents a work in progress. \
+        If it does not work, check back in a few days.  Or try \
+        deactivating the currently active gnuradio port, \
+        cleaning any current builds, and trying again.
+
+    conflicts           gnuradio gnuradio37
+
+    set python_versions { 2.7 3.5 3.6 3.7 3.8 }
+    # TODO I choose this version because all LTS Linux use that
+    set default_python_variant +python36
 
     # temporary patch to fix and allow external setting for
     # CMAKE_C/CXX_STANDARD
-
     patchfiles-append \
-        patch-C_CXX_STANDARD.next.diff
-
-    # temporary patch to fix using Thrift 0.9.2 through 0.12.0
-
+        patch-C_CXX_STANDARD.38.diff
```

Comment:
guessing some the .38 patches will fail here once the port version is updated to current ... let's keep the "next" patches as .next